src/services/webclient.ts

```diff
@@ -2008,12 +2008,39 @@ export class WebClientService {
             .then((titlePrefix) =>  {
                 const title = `${titlePrefix} ${senderName}`;
                 let body = '';
-                switch (message.type) {
+                let messageType = message.type;
+                let caption = message.caption;
+                let captionString = '';
+                if (typeof caption !== 'undefined') {
+                    captionString = captionString + ': ' + caption;
+                }
+                let messageTypeString = this.$translate.instant('messageTypes.' + messageType);
+                switch (messageType) {
                     case 'text':
                         body = message.body;
                         break;
+                    case 'location':
+                        body = messageTypeString + ': ' + message.location.poi;
+                        break;
+                    case 'file':
+                        if (message.file.type === 'image/gif') {
+                            body = this.$translate.instant('messageTypes.' + 'gif') + captionString;
+                            break;
+                        }
+                        // Display caption, if available otherwise use filename
+                        if (captionString.length > 0) {
+                            body = messageTypeString + captionString;
```

Comment:
It´s not missing it´s outsourced :)

On line 2013 `captionString` is initialized as an empty string. In the next few lines if the media has a caption it is added along with `': '`. The `captionString` will look like `: Your caption here` and the body will be something like `Image: Your caption here`.
If there is no caption it will be `Image`.